quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/hostname/DefaultHostnameProvider.java

```diff
@@ -0,0 +1,211 @@
+        return noProxy ? defaultTlsPort : requestPort;
+    }
+
+    private <T> T forNonStrictBackChannel(UriInfo originalUriInfo, UrlType urlType,
+            BiFunction<UriInfo, UrlType, T> frontEndTypeResolver, Function<UriInfo, T> defaultResolver) {
+        if (BACKEND.equals(urlType) && !strictBackChannel) {
+            if (isHostFromFrontEndUrl(originalUriInfo)) {
+                return frontEndTypeResolver.apply(originalUriInfo, FRONTEND);
+            }
+
+            return defaultResolver.apply(originalUriInfo);
+        }
+
+        return null;
+    }
+
+    private <T> T fromFrontChannel(UriInfo originalUriInfo, Function<URI, T> frontEndTypeResolver, Function<UriInfo, T> defaultResolver,
+            T defaultValue) {
+        URI frontEndUrl = getRealmFrontEndUrl();
+
+        if (frontEndUrl != null) {
+            return frontEndTypeResolver.apply(frontEndUrl);
+        }
+
+        return defaultValue == null ? defaultResolver.apply(originalUriInfo) : defaultValue;
+
+    }
+
+    private boolean isHostFromFrontEndUrl(UriInfo originalUriInfo) {
+        String requestHost = getHostname(originalUriInfo);
+        String frontendUrl = getHostname(originalUriInfo, FRONTEND);
+
+        if (requestHost.equals(frontendUrl)) {
+            return true;
+        }
+
+        URI realmUrl = getRealmFrontEndUrl();
+
+        return realmUrl != null && requestHost.equals(realmUrl.getHost());
+    }
+
+    protected URI getRealmFrontEndUrl() {
+        KeycloakSession session = Resteasy.getContextData(KeycloakSession.class);
+        URI realmUrl = (URI) session.getAttribute(REALM_URI_SESSION_ATTRIBUTE);
+
+        if (realmUrl == null) {
+            RealmModel realm = session.getContext().getRealm();
+
+            if (realm != null) {
+                String frontendUrl = realm.getAttribute("frontendUrl");
+
+                if (isNotBlank(frontendUrl)) {
+                    realmUrl = URI.create(frontendUrl);
+                    session.setAttribute(DefaultHostnameProvider.REALM_URI_SESSION_ATTRIBUTE, realmUrl);
+                    return realmUrl;
+                }
+            }
+        }
+
+        return realmUrl;
+    }
+
+    @Override
+    public void close() {
+
+    }
+
+    @Override
+    public String getId() {
+        return "default";
+    }
+
+    @Override
+    public HostnameProvider create(KeycloakSession session) {
+        return this;
+    }
+
+    @Override
+    public void init(Config.Scope config) {
+        frontChannelHostName = config.get("hostname");
+
+        if (config.getBoolean("strict", false) && frontChannelHostName == null) {
+            throw new RuntimeException("Strict hostname resolution configured but no hostname was set");
+        }
+
+        Boolean strictHttps = config.getBoolean("strict-https", false);
+
+        if (strictHttps) {
+            defaultHttpScheme = "https";
+        }
+
+        defaultPath = config.get("path");
```

Comment:
Shouldn't this get the context-path of Keycloak if not set? Atm that's root, but we want that to be configurable right?

Reply:
looks like it's defaulting to the context path from the request - I think that's fine, and may be better to defaulting to the context-path of KC

Reply:
Yeah, the root path is now configurable via `http.relative-path` option. And right now the provider defaults to request path.

I'm not 100% sure about this path option, perhaps I'm still missing the use case. Otherwise, I would just remove it for now.